Network/Public/Include/Network/iAPI.hpp

```diff
@@ -0,0 +1,184 @@
+            void messageAllClients              (std::shared_ptr<Network::Connection> exceptionConnection,const Message& message) const override;   \
+            void checkServerPing                (std::shared_ptr<Network::Connection> clientConnection,   const Message& message) const override;   \
+            void readAllMessage                 (std::shared_ptr<Network::Connection> clientConnection,   const Message& message) const override;   \
+            void channelListRequest             (std::shared_ptr<Network::Connection> clientConnection) const override;                             \
+            void messageHistoryRequest          (std::shared_ptr<Network::Connection> clientConnection,   const Message& message) const override;   \
+            void messageStoreRequest            (std::shared_ptr<Network::Connection> clientConnection,   const Message& message) const override;   \
+            void replyHistoryRequest            (std::shared_ptr<Network::Connection> clientConnection,   const Message& message) const override;   \
+            void replyStoreRequest              (std::shared_ptr<Network::Connection> clientConnection,   const Message& message) const override;   \
+            void messageDeleteRequest           (std::shared_ptr<Network::Connection> clientConnection,   const Message& message) const override;   \
+            void messageEditRequest             (std::shared_ptr<Network::Connection> clientConnection,   const Message& message) const override;   \
+            void messageReactionRequest         (std::shared_ptr<Network::Connection> clientConnection,   const Message& message) const override;   \
+            void registrationRequest            (std::shared_ptr<Network::Connection> clientConnection,   const Message& message) const override;   \
+            void loginRequest                   (std::shared_ptr<Network::Connection> clientConnection,   const Message& message) const override;   \
+            void channelLeaveRequest            (std::shared_ptr<Network::Connection> clientConnection,   const Message& message) const override;   \
+            void channelSubscribeRequest        (std::shared_ptr<Network::Connection> clientConnection,   const Message& message) const override;   \
+            void channelSubscriptionListRequest (std::shared_ptr<Network::Connection> clientConnection) const override;                             \
+            void channelDeleteRequest           (std::shared_ptr<Network::Connection> clientConnection,   const Message& message) const override;   \
+            void channelCreateRequest           (std::shared_ptr<Network::Connection> clientConnection,   const Message& message) const override;   \
+            void directMessageCreateRequest     (std::shared_ptr<Network::Connection> clientConnection,   const Message& message) const override;   \
+            void defaultRequest                 () const override;
+
+
+
+
+
+namespace Network{
+    class iAPI
```

Comment:
Which API is it? I would mark specifically what kind of API it is. (conventionally speaking ClientAPI).
On line 7 you use using ClientConn. So it's a ClientAPI?